### ISSUE TITLE
request delete of old MVI/MPI PS objects

### ISSUE THREAD

**Issue by joeniquette** [department-of-veterans-affairs/va.gov-team#69275]
The following should be removed from PS related to the most recent MPI Cert changes. We were able to do this because we consolidated the references to the certs to fewer values. We also reduced our lower environment certs to a single cert (down from 4). Finally we removed the PS value pointers to cert paths that can just be hardcoded paths inside the config files. We did not change anything for production yet but we will in the next ticket.

1. /devops/certificates/mvi_int_cert_dev.pem
2. /devops/certificates/mvi_int_dev.key
3. /devops/certificates/mvi_qa_cert_dev.pem
4. /devops/certificates/mvi_qa_dev.key
5. /dsva-vagov/fwdproxy/int/mvi_key
6. /dsva-vagov/fwdproxy/pint/mvi_key
7. /dsva-vagov/fwdproxy/preprod/mvi_key
8. /dsva-vagov/fwdproxy/qa/mvi_key
9. /dsva-vagov/identity-team/fwdproxy/int/mvi_key
10. /dsva-vagov/identity-team/fwdproxy/preprod/mvi_key
11. /dsva-vagov/identity-team/fwdproxy/qa/mvi_key
12. /dsva-vagov/identity-team/lowers/mvi_qa_cert_dev.pem
13. /dsva-vagov/identity-team/lowers/vetsgov-mvi-int-cert.pem
14. /dsva-vagov/identity-team/lowers/vetsgov-mvi-preprod-cert.pem
15. /dsva-vagov/vets-api/common/mvi/client_cert_path
16. /dsva-vagov/vets-api/common/mvi/client_key_path
17. /dsva-vagov/vets-api/common/mvi/hca_client_key_path
18. /dsva-vagov/vets-api/common/mvi_hca/client_cert_path
19. /dsva-vagov/vets-api/dev/mvi/client_cert_path
20. /dsva-vagov/vets-api/dev/mvi_hca/url
21. /dsva-vagov/vets-api/dev/vetsgov-mvi-int-cert.pem
22. /dsva-vagov/vets-api/dev/vetsgov-mvi-qa-cert.pem
23. /dsva-vagov/vets-api/prod/mvi/client_key_path
24. /dsva-vagov/vets-api/prod/mvi_hca/url
25. /dsva-vagov/vets-api/sandbox/mvi/client_key_path
26. /dsva-vagov/vets-api/sandbox/mvi/hca_client_key_path
27. /dsva-vagov/vets-api/sandbox/mvi/url
28. /dsva-vagov/vets-api/sandbox/mvi_hca/client_cert_path
29. /dsva-vagov/vets-api/sandbox/mvi_hca/url
30. /dsva-vagov/vets-api/staging/mvi_hca/url
31. /dsva-vagov/vets-api/staging/vetsgov-mvi-int-cert.pem
32. /dsva-vagov/vets-api/staging/vetsgov-mvi-preprod-cert.pem
33. /dsva-vagov/vets-api/staging/vetsgov-mvi-preprod.key
34. /dsva-vagov/vets-api/staging/vetsgov-mvi-qa-cert.pem

**Comment by joeniquette** [comment]
I also retired the int, preprod, and pint certs in venafi for this connection.

**Comment by ph-One** [comment]
@barbarello may be interested in this work

**Comment by barbarello** [comment]
looking into this

**Comment by barbarello** [comment]
Manually deleted 6 (managed) via terraform and 21 (unmansged) via aws console.

**Comment by barbarello** [comment]
turns out #20 (/dsva-vagov/vets-api/sandbox/mvi/url) should not haven listed in the first place as it is being used in https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/

**Comment by rmtolmach** [comment]
jinx @barbarello 😆

**Comment by barbarello** [comment]
@rmtolmach it wouldn't make sense to delete it from parameter store if pods are using it.

**Comment by rmtolmach** [comment]
@barbarello the value would need to be deleted from [here](https://github.com/search?q=repo%3Adepartment-of-veterans-affairs%2Fvsp-infra-application-manifests%20%2Fdsva-vagov%2Fvets-api%2Fsandbox%2Fmvi%2Furl&type=code) first and then deleted from Param store, if that's what Joe wants. Or we just leave it in both places. The scenario yesterday was that it was only removed from PS but it was still in the manifest.

**Comment by barbarello** [comment]
@rmtolmach agreed on the cleanup process.   What i'm trying to communicate here is that if there is an operational need for a key,  which is from a pod's perspective,   it should not be deleted from paramter store.       
So , just saying keys are no longer needed is not enough.

**Comment by rmtolmach** [comment]
@barbarello got it. Looks like it is used https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/mpi/configuration.rb#L25

https://github.com/search?q=repo%3Adepartment-of-veterans-affairs%2Fvets-api%20MPI%3A%3AService.new&type=code

**Comment by barbarello** [comment]
unchecked items 1. through 8.  as they are used in https://github.com/department-of-veterans-affairs/devops

**Comment by joeniquette** [comment]
@rmtolmach it looks like @barbarello isnt on the project team anymore. Is this something someone else can finish?

**Comment by rmtolmach** [comment]
👋 Passing along to the DevOps CoP lead @pjhill

**Comment by jwoodman5** [comment]
@pjhill to add to DevOps CoP meeting agenda 2/15.

**Comment by pjhill** [comment]
@stoiven -- will address this as he has time during the current support rotation and hand off any remaining items when the next devops support person comes on

**Comment by pjhill** [comment]
No updates in a while -- I'll need to dig up an appropriate summary of recent activity.

**Comment by joeniquette** [comment]
@pjhill just doing my monthly check-in on this ticket, any updates?

**Comment by pjhill** [comment]
Parameter store and certificate contained within it are being cleaned up as a low priority item as members of DevOps COP scattered across various Platform teams come into contact with aspects of it. I will bring up this list in the DevOps COP meeting this Thursday -- 6/27.
